homeassistant/components/switcher_kis/utils.py

```diff
@@ -53,3 +55,9 @@ def on_device_data_callback(device: SwitcherBase) -> None:
 
     _LOGGER.debug("Finished discovery, discovered devices: %s", len(discovered_devices))
     return discovered_devices
+
+
+@singleton.singleton("switcher_breeze_remote_manager")
+def get_breeze_remote_manager(hass: HomeAssistant) -> SwitcherBreezeRemoteManager:
+    """Get Switcher Breeze remote manager."""
+    return SwitcherBreezeRemoteManager()
```

Comment:
Side effects on instantiation like that in the library is not nice.

Reply:
The library doesn't care if you create multiple instances, this is used to make HA more efficient since the remote manager access a file to read remotes codes and cache them in memory. Holding a single instance helps for situations that the file is already cached in memory. The reason I am doing it here is that I want to share the same instance between the `climate` and `button` platforms, since they will also need the same remote type so it is better to hold a single instance.

Reply:
My comment is not about the singleton code here. It's about the side effect in the library. It checks if a file exists. That's not nice to do at this stage. Side effects should be done in explicit function calls.

Reply:
Thanks, I didn't notice we raise at the library, I will remove this check, [get_remote](https://github.com/TomerFi/aioswitcher/blob/611f82437e70aa63c8b97f1591bdd38ec7f0ceb0/src/aioswitcher/api/remotes.py#L440) will raise `FileNotFoundError` if the file is missing, which should never happen anyhow for Home assistant since we use the default file which is prepacked with the library.

Reply:
Thanks! Sorry, I was unclear with my first comment.